### ISSUE TITLE
Modal - Ability to modify z-index

### ISSUE THREAD

**Issue by davidwparker** [angular-ui/bootstrap#3580]
@rvanbaalen - I'm opening a new issue for this, since you locked the other one. I'm not satisfied with you locking it, especially after I made a clear comment on the issue, which was never addressed and still isn't addressed- back in November 2014 (https://github.com/angular-ui/bootstrap/issues/2373#issuecomment-63193732). 

Here's the comment / concern, in case you didn't want to click the link:

"this only works if you have one type of modal. What should I do if I have multiple types where one particular modal needs to use a different template/z-index?

A configurable option is much better, but I don't know how to go about adding one (or choosing a different template)..."

If you have a solution to that, feel free to close this issue too. The current solution of `templateUrl` + `windowClass` doesn't work.

**Comment by rvanbaalen** [comment]
@davidwparker I locked the issue mainly because it was flooded by '+1' comments months after it got closed.

As for all issues -- demonstrate your problem by posting a plunkr and we can take a look at it.

**Comment by davidwparker** [comment]
@rvanbaalen - I'll see what I can do. We had to customize Angular-UI to get our modals working and haven't touched/upgraded it since. Thanks and agreed that +1 isn't useful.

**Comment by pkozlowski-opensource** [comment]
@davidwparker it would help if you could present a real-life use-case instead of general comments like 

> What should I do if I have multiple types where one particular modal needs to use a different template/z-index

Do you have this use case? How often? What are the alternative solutions we could have?

The problem with adding everything for every use case is that:
* API is growing
* library size is growing

So we need to weight those factors and having details on your use-case help with this.

**Comment by davidwparker** [comment]
@pkozlowski-opensource - our use case is we have 4x different modal types within our application. Two of the modals are heavily customized (think modal-style dropdowns like Facebook's notifications / messages), one is more of a general purpose modal (using mostly defaults), and the last is like a general purpose modal, but with a few minor changes.

I'll see about putting together a plunkr so you can see some examples.

**Comment by rvanbaalen** [comment]
And there is a situation in your use case where multiple modals are open at the same time and thus you need to modify it's z-indexes?

**Comment by davidwparker** [comment]
@rvanbaalen - exactly right.

**Comment by rvanbaalen** [comment]
I simply think you're using a hammer to drill a screw in the wall if you know what I mean. Consider this definition of a modal:

> A modal window is any type of window that is a child (secondary window) to a parent window and usurps the parent's control. It is commonly associated with an Internet Web site pop-up window that stays in front of the original window.  **A user may not press any controls or enter any information on the parent window (the original window which opened the modal) until the modal has been closed.** (*hence the z-index*) A modal window is commonly used when the author wants to retain the user's focus on the information in the modal as it is impossible for the user to interact with the other windows of the same process.

I think your implementation needs something other than multiple modals with different z-indexes.

**Comment by davidwparker** [comment]
@rvanbaalen - interesting. I think it's a matter of trying to not recreate the wheel, and using something that does what it needs to 99% of the time. Consider the FB examples again, they're somewhere between a lesser modal / a super dropdown. For our particular application, we figured that the functionality provided by angular-ui modal fulfilled more of what we needed to do to customize that the of angular-ui dropdown. TBH, I feel like it's 6 of one, have dozen of another.
